metadata/containers.go

```diff
@@ -71,6 +82,48 @@ func (s *containerStore) List(ctx context.Context, filter string) ([]containers.
 	return m, nil
 }
 
+func adaptContainer(o interface{}) filters.Adaptor {
+	obj := o.(containers.Container)
+	return filters.AdapterFunc(func(fieldpath []string) (string, bool) {
+		if len(fieldpath) == 0 {
+			return "", false
+		}
+
+		switch fieldpath[0] {
```

Comment:
Shouldn't we loop upon all fieldpath values?

What if I have `runtime==linux,labels==foo` as my filter?